src/sidepanel/v2/components/ProviderSettingsCard.tsx

```
@@ -0,0 +1,426 @@
+type ProviderDraft = z.infer<typeof ProviderFormSchema>
+
+export function ProviderSettingsCard(): JSX.Element | null {
+  const { sendMessage, addMessageListener, removeMessageListener } = useSidePanelPortMessaging()
+  const [config, setConfig] = useState<BrowserOSProvidersConfig | null>(null)
+  const [draftOpen, setDraftOpen] = useState<boolean>(false)
+  const [draft, setDraft] = useState<ProviderDraft | null>(null)
+  const [saving, setSaving] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+  const [selectedProviderId, setSelectedProviderId] = useState<string | null>(null)
+  const [editingProviderId, setEditingProviderId] = useState<string | null>(null)
+  const [missing, setMissing] = useState<{ name: boolean; apiKey: boolean; modelId: boolean; type: boolean }>({ name: false, apiKey: false, modelId: false, type: false })
```

ESLint

[error] 'missing' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)